src/actions/guides/database/deleting_records.cr

```diff
@@ -0,0 +1,194 @@
+    * First, we need to add a new `soft_deleted_at : Time?` column to the table that needs soft deletes.
+
+    ```
+    # Run this in your terminal
+    lucky gen.migration AddSoftDeleteToArticles
+    ```
+
+    * Open your new `db/migrations/#{Time.utc.to_s("%Y%m%d%H%I%S")}_add_soft_delete_to_articles.cr` file.
+
+    ```crystal
+    def migrate
+      alter table_for(Article) do
+        add soft_deleted_at : Time?
+      end
+    end
+    ```
+
+    * Now open your `src/models/article.cr` file.
+
+    ```crystal
+    class Article < BaseModel
+      # Include this module to add methods for
+      # soft deleting and restoring
+      include Avram::SoftDelete::Model
+
+      table do
+        # Add the new column to your model
+        column soft_deleted_at : Time?
+      end
+    end
+    ```
+
+    * Next you need to update `src/queries/article_query.cr`.
+
+    ```crystal
+    class ArticleQuery < Article::BaseQuery
+      # Include this module to add methods for
+      # querying and soft deleting records
+      include Avram::SoftDelete::Query
+    end
+    ```
+
+    ### Marking a record as soft deleted
+
+    Your model instance will now have a `soft_delete` method to mark that record as soft deleted, as well as,
+    a `soft_deleted?` method to check if the record has been marked as soft deleted.
+
+    ```crystal
+    article = ArticleQuery.first
+
+    # Check to see if soft_deleted_at is present
+    article.soft_deleted? #=> false
+
+    # Save the record as soft deleted
+    article.soft_delete
+
+    # Reload the model to get the new value
+    article.reload.soft_deleted? #=> true
+    ```
+
+    ### Soft deleting in bulk
+
+    You can bulk update a group of records as soft deleted with the `soft_delete` method on your Query object.
+
+    ```crystal
+    articles_to_delete = ArticleQuery.new.created_at.gt(3.years.ago)
+
+    # Marks the articles created over 3 years ago as soft deleted
+    articles_to_delete.soft_delete
+    ```
+
+    ### Restore a soft deleted record
+
+    If you need to restore a soft deleted record, you can use the `restore` method.
+
+    ```crystal
+    # Set the `soft_deleted_at` back to `nil`
+    article.restore
+    ```
+
+    ### Bulk restoring soft deleted records
+
+    The same as we can bulk soft delet records, we can also bulk update to restore them with
+    the `restore` method.
+
+    ```crystal
+    articles_to_restore = ArticleQuery.new.published_at.lt(1.week.ago)
+
+    # Restore recently published articles
+    articles_to_restore.restore
+    ```
+
+    ### Query soft deleted records
+
+    ```crystal
+    # Return all articles that are not soft deleted
+    ArticleQuery.new.only_kept
+
+    # Return all articles that are soft deleted
+    ArticleQuery.new.only_soft_deleted
+    ```
+
+    ### Default queries without soft deleted
+
+    If you want to skip all of your soft deleted records by default, it's really easy to do.
+    Just add the `only_kept` method to your `initialize`.
```

Comment:
```suggestion
    Just add the `only_kept` method to an `initialize` method.
```

Changed this since by default there isn't one set so might be confusing.